packages/headless/src/controllers/commerce/core/facets/category/headless-commerce-category-facet.test.ts

```diff
@@ -0,0 +1,307 @@
+import {
+  CategoryFacetValueRequest,
+  CommerceFacetRequest,
+} from '../../../../../features/commerce/facets/facet-set/interfaces/request';
+import {CategoryFacetValue} from '../../../../../features/commerce/facets/facet-set/interfaces/response';
+import {
+  toggleSelectCategoryFacetValue,
+  updateCategoryFacetNumberOfValues,
+} from '../../../../../features/facets/category-facet-set/category-facet-set-actions';
+import {CommerceAppState} from '../../../../../state/commerce-app-state';
+import {buildMockCommerceFacetRequest} from '../../../../../test/mock-commerce-facet-request';
+import {buildMockCategoryFacetResponse} from '../../../../../test/mock-commerce-facet-response';
+import {buildMockCommerceFacetSlice} from '../../../../../test/mock-commerce-facet-slice';
+import {buildMockCategoryFacetValue} from '../../../../../test/mock-commerce-facet-value';
+import {buildMockCommerceState} from '../../../../../test/mock-commerce-state';
+import {
+  MockedCommerceEngine,
+  buildMockCommerceEngine,
+} from '../../../../../test/mock-engine-v2';
+import {commonOptions} from '../../../product-listing/facets/headless-product-listing-facet-options';
+import {
+  CategoryFacet,
+  CategoryFacetOptions,
+  buildCategoryFacet,
+} from './headless-commerce-category-facet';
+
+jest.mock(
+  '../../../../../features/facets/category-facet-set/category-facet-set-actions'
+);
+
+describe('CategoryFacet', () => {
+  const facetId: string = 'category_facet_id';
+  let engine: MockedCommerceEngine;
+  let state: CommerceAppState;
+  let options: CategoryFacetOptions;
+  let facet: CategoryFacet;
+
+  function initEngine(preloadedState = buildMockCommerceState()) {
+    engine = buildMockCommerceEngine(preloadedState);
+  }
+
+  function initCategoryFacet() {
+    facet = buildCategoryFacet(engine, options);
+  }
+
+  function setFacetState(
+    config: Partial<CommerceFacetRequest<CategoryFacetValueRequest>> = {},
+    moreValuesAvailable = false
+  ) {
+    state.commerceFacetSet[facetId] = buildMockCommerceFacetSlice({
+      request: buildMockCommerceFacetRequest({
+        facetId,
+        type: 'hierarchical',
+        ...config,
+      }),
+    });
+    state.productListing.facets = [
+      buildMockCategoryFacetResponse({
+        moreValuesAvailable,
+        facetId,
+        type: 'hierarchical',
+        values: (config.values as CategoryFacetValue[]) ?? [],
+      }),
+    ];
+  }
+
+  // eslint-disable-next-line @cspell/spellchecker
+  // TODO CAPI-90: Test facet search
```

Comment:
This PR is quite big already. I'll tackle adding facet search to category facets in a follow-up PR.